app/frontend/actions/wallet.ts

```diff
@@ -23,18 +30,24 @@ const getShouldShowSaturatedBanner = (accountsInfo: Array<AccountInfo>) =>
   accountsInfo.some(({poolRecommendation}) => poolRecommendation.shouldShowSaturatedBanner)
 
 type Wallet = ReturnType<typeof ShelleyWallet>
-let wallet: Wallet
-export const setWallet = (w: Wallet) => {
+let wallet: Wallet | null
+export const setWallet = (w: Wallet | null) => {
   wallet = w
 }
-export const getWallet = (): Wallet => wallet
+export const getWallet = (): Wallet | null => wallet
 
+export const getWalletOrThrow = (): Wallet => {
```

Comment:
I see that getWallet() is not used anymore anywhere else, so I'd  rather rename `getWalletOrThrow()` to `getWallet()` and get rid of the old `getWallet()` function that wasn't checking whether the wallet is ready as the `OrThrow` suffix is arguably just exposing a lower-level concern than the callers would care about most of the time